homeassistant/components/mqtt/client.py

```diff
@@ -309,9 +310,10 @@ def setup(self) -> None:
             clean_session = True
 
         if (client_id := config.get(CONF_CLIENT_ID)) is None:
-            # PAHO MQTT relies on the MQTT server to generate random client IDs.
-            # However, that feature is not mandatory so we generate our own.
-            client_id = None
+            # PAHO MQTT relies on the MQTT server to generate random client ID
+            # for protocol version 3.1, however, that feature is not mandatory
+            # so we generate our own.
+            client_id = mqtt._base62(uuid4().int, padding=22)  # noqa: SLF001
```

Comment:
I guess there isn't a good way to avoid the protected _base62 call.. probably still better than using another lib, and any breakage will be caught with the test

Reply:
Sure, it used to be globally exposed, an I agree this should be avoided. I suggest to repair this in a follow up PR.